vtm/src/org/oscim/core/GeometryBuffer.java

```diff
@@ -468,6 +468,22 @@ public float isClockwise() {
         return GeometryUtils.isClockwise(points, index[0]);
     }
 
+    /**
+     * Remove the last point
+     */
+    public void removeLastPoint() {
+        if (isTris()) {
+            int indexSize = 0;
+            while (indexSize < index.length && index[indexSize] != -1) {
+                indexSize++;
+            }
+            index[indexSize - 3] = -1;
```

Comment:
Only `index` needs changes?

Reply:
Yes, points could be referenced by lower indices, so wouldn't be good to remove them.